### PR TITLE
Feature: Added new deployment toggles to hub-spoke

### PR DESCRIPTION
**Pull request by oZakari**
# Overview/Summary
Add additional deployment toggles for the hub-spoke module for feature parity with the VWAN module.

## This PR fixes/adds/changes/removes

1. Added VPN and ER gateway deployment toggles

### Breaking Changes

1. None

## Testing Evidence

Validated the following with both parVpnGatewayConfig and parExpressRouteGatewayConfig objects containing data:

- When testing with parVpnGatewayEnabled to true and parExpressRouteGatewayEnabled. Only VPN GW resources provisioned.
- When testing with parVpnGatewayEnabled to false and parExpressRouteGatewayEnabled false. Only ER GW resources provisioned.
- When both params set to true both sets of resources created.
- When both params set to false, both sets of resources not provisioned.

## As part of this Pull Request I have

- [x] Read the [Contribution Guide](https://github.com/Azure/ALZ-Bicep/wiki/Contributing) and ensured this PR is compliant with the guide
- [x] Ensured the resource API versions in `.bicep` file/s I am adding/editing are using the latest API version possible
- [x] Checked for duplicate [Pull Requests](https://github.com/Azure/ALZ-Bicep/pulls)
- [x] Associated it with relevant [GitHub Issues](https://github.com/Azure/ALZ-Bicep/issues)
- [x] *(ALZ Bicep Core Team Only)* Associated it with relevant [ADO Items](https://aka.ms/alz/bicep/backlog)
- [x] Ensured my code/branch is up-to-date with the latest changes in the `main` [branch](https://github.com/Azure/ALZ-Bicep/tree/main)
- [x] Performed testing and provided evidence.
- [x] Updated one or more of the following tests *(if required)*
  - [Unit](https://github.com/Azure/ALZ-Bicep/blob/main/.github/workflows/bicep-build-to-validate.yml)
  - [Linting](https://github.com/Azure/ALZ-Bicep/tree/main/.github/workflows)
  - [E2E (End-To-End)](https://github.com/Azure/ALZ-Bicep/blob/main/tests/pipelines/bicep-build-to-validate.yml)
  - [ValidateAzCloud (Base validation in Azure Cloud)](https://github.com/Azure/ALZ-Bicep/blob/main/tests/pipelines/base-unit-validate.yml)
  - [ValidateMcCloud (Base validation in Azure China Cloud)](https://github.com/Azure/ALZ-Bicep/blob/main/tests/pipelines/mc-base-unit-validate.yml)
- [x] Updated relevant and associated documentation (e.g. Contribution Guide, Module READMEs, Wiki Docs etc.)
- [x] If relevant, created or updated Code Tours [here](https://github.com/Azure/ALZ-Bicep/blob/main/.vscode/tours)

### CODE REVIEW

**Comment by FallenHoot** [comment]
The following is missing to make this work as it is now. Without this, it will check if a subnet is created with `GatewaySubnet`
```
resource resGatewaySubnetRef 'Microsoft.Network/virtualNetworks/subnets@2023-02-01' existing = if (parVpnGatewayEnabled || parExpressRouteGatewayEnabled) {
  parent: resHubVnet
  name: 'GatewaySubnet'
}
```

I also noticed that varVpnGwConfig and varErGwConfig will deploy the PublicIP by default. I want to leave the code in for GwConfig, because maybe I want to use it later. I only care if the trigger/toggle/boolean is false. If it is false, then it gives the `json('{"name": "noconfigVpn"}')`. Resulting in Gateway Public IP to not being created. 

```
var varVpnGwConfig = ((parVpnGatewayEnabled) ? parVpnGatewayConfig : json('{"name": "noconfigVpn"}'))

var varErGwConfig = ((parExpressRouteGatewayEnabled) ? parExpressRouteGatewayConfig : json('{"name": "noconfigEr"}'))
```

Also, a minor wording issue:
```
@sys.description('Switch to enable/disable VPN virtual network gateway deployment.')
param parExpressRouteGatewayEnabled bool = true
```
Should say.
```
@sys.description('Switch to enable/disable ExpressRoute virtual network gateway deployment.')
param parExpressRouteGatewayEnabled bool = true
```

**Comment by oZakari** [comment]
Thanks @FallenHoot, as mentioned in Teams thread, going to open separate PR for subnet related changes. Have included your other recommendations though.

File: infra-as-code/bicep/modules/hubNetworking/hubNetworking.bicep [hunk 1]
```diff
@@ -332,9 +332,9 @@ var varSubnetProperties = [for subnet in varSubnetMap: {
   }
 }]
 
-var varVpnGwConfig = ((!empty(parVpnGatewayConfig)) ? parVpnGatewayConfig : json('{"name": "noconfigVpn"}'))
+var varVpnGwConfig = ((parVpnGatewayEnabled) ? parVpnGatewayConfig : json('{"name": "noconfigVpn"}'))
```
Comment (jtracey93) [inline infra-as-code/bicep/modules/hubNetworking/hubNetworking.bicep:335]:
```suggestion
var varVpnGwConfig = ((parVpnGatewayEnabled) &&(!empty(parVpnGatewayConfig)) ? parVpnGatewayConfig : json('{"name": "noconfigVpn"}'))
```

should we keep this logic to check the block isnt empty? i've added it as an AND but do we think it should be an OR?

**Review by jtracey93 — CHANGES_REQUESTED** [review]
Just a couple of comments

File: infra-as-code/bicep/modules/hubNetworking/hubNetworking.bicep (hunk 1 above)
Comment (oZakari) [reply infra-as-code/bicep/modules/hubNetworking/hubNetworking.bicep:335]: Good point, I think it makes sense as you recommended.

**Pull request merged**